### PR TITLE
[Merge-Queue] Validate reviewers in commit message

### PR DESCRIPTION
**Pull request by JonWBedard**
#### c2855b1b43fbcb079d4947375ec3472001ee323f
<pre>
[Merge-Queue] Validate reviewers in commit message
<a href="https://bugs.webkit.org/show_bug.cgi?id=240718">https://bugs.webkit.org/show_bug.cgi?id=240718</a>
&lt;rdar://problem/93665771 &gt;

Reviewed by Aakash Jain.

* Tools/CISupport/ews-build/steps.py:
(ValidateCommitMessage):
(ValidateCommitMessage.__init__):
(ValidateCommitMessage.extract_reviewers): Extract reviewers from commit message.
(ValidateCommitMessage.is_reviewer): Check if a contributor is a reviewer.
(ValidateCommitMessage.run): Check if extracted reviewers are reviewers.
* Tools/CISupport/ews-build/steps_unittest.py:

Canonical link: <a href="https://commits.webkit.org/250819@main">https://commits.webkit.org/250819@main</a>
git-svn-id: <a href="https://svn.webkit.org/repository/webkit/trunk@294582">https://svn.webkit.org/repository/webkit/trunk@294582</a> 268f45cc-cd09-0410-ab3c-d52691b4dbfc
</pre>

### CODE REVIEW

File: Tools/CISupport/ews-build/steps.py [hunk 1]
```diff
@@ -5055,15 +5079,32 @@ def run(self, BufferLogObserverClass=logobserver.BufferLogObserver):
...
+                self.summary = "Failed to load contributors.json, can't validate reviewers"
+                rc = FAILURE
+            elif reviewers and not all([self.is_reviewer(reviewer) for reviewer in reviewers])
+                self.summary = '{} is not a reviewer'
+                for reviewer in reviewers:
+                    if not self.is_reviewer(reviewer):
+                        self.summary = self.summary.format(reviewer)
+                        break
+                rc = FAILURE
+            else:
```
Comment (aj062) [inline Tools/CISupport/ews-build/steps.py]: what if reviewers is None?

**Review by aj062 — APPROVED** [review]

File: Tools/CISupport/ews-build/steps.py (hunk 1 above)
Comment (JonWBedard) [reply Tools/CISupport/ews-build/steps.py]: That means you have some variant of "Unreviewed", or you wouldn't have passed the 2nd check in this step.

Comment (aj062) [reply Tools/CISupport/ews-build/steps.py]: ok

**Comment by aj062** [comment]
Please have a look at failing unit-tests as well.

**Comment by JonWBedard** [comment]
> Please have a look at failing unit-tests as well.

Yep! That's why I made this a draft, would like to add a unit test as well.

**Pull request merged**